bin/alluxio-start.sh

```diff
@@ -52,8 +52,7 @@ check_mount_mode() {
     Mount);;
     SudoMount);;
     NoMount)
-      mount | grep "${ALLUXIO_RAM_FOLDER}" > /dev/null
-      if [[ $? -ne 0 || -z ${ALLUXIO_RAM_FOLDER} ]]; then
+      if [[ $(mount | grep -c ${ALLUXIO_RAM_FOLDER}) != 0 ]]; then
```

Comment:
@pfxuan  You made a good point. I didn't think of a user who wants to mount the directory by himself/herself and then use a sub-directory of that as ALLUXIO_RAM_FOLDER. Personally, i would prefer to change the documentation and var name. But you made a good point about backward compatibility. 

How about this:
Write a function to detect whether ramfs or tmpfs is mounted on ALLUXIO_RAM_FOLDER or its parents. 

** Note that your fix won't work if the ALLUXIO_RAM_FOLDER matches the mount point.